lib/pbench/test/unit/server/test_datasets_contents.py

```diff
@@ -0,0 +1,278 @@
+from http import HTTPStatus
+from pathlib import Path
+from typing import Optional
+
+import pytest
+import requests
+
+from pbench.server.cache_manager import BadDirpath, CacheManager, CacheObject, CacheType
+from pbench.server.database.models.datasets import Dataset, DatasetNotFound
+
+
+class TestDatasetsAccess:
+    @pytest.fixture()
+    def query_get_as(self, client, server_config, more_datasets, pbench_drb_token):
+        """
+        Helper fixture to perform the API query and validate an expected
+        return status.
+
+        Args:
+            client: Flask test API client fixture
+            server_config: Pbench config fixture
+            more_datasets: Dataset construction fixture
+            pbench_drb_token: Authenticated user token fixture
+        """
+
+        def query_api(
+            dataset: str, target: str, expected_status: HTTPStatus
+        ) -> requests.Response:
+            try:
+                dataset_id = Dataset.query(name=dataset).resource_id
+            except DatasetNotFound:
+                dataset_id = dataset  # Allow passing deliberately bad value
+            headers = {"authorization": f"bearer {pbench_drb_token}"}
+            response = client.get(
+                f"{server_config.rest_uri}/datasets/{dataset_id}/contents/{target}",
+                headers=headers,
+            )
+            assert (
+                response.status_code == expected_status
+            ), f"Unexpected failure '{response.json}'"
+            return response
+
+        return query_api
+
+    def test_get_no_dataset(self, query_get_as):
+        response = query_get_as(
+            "nonexistent-dataset", "metadata.log", HTTPStatus.NOT_FOUND
+        )
+        assert response.json == {"message": "Dataset 'nonexistent-dataset' not found"}
+
+    def test_dataset_not_present(self, query_get_as):
+        response = query_get_as("fio_2", "metadata.log", HTTPStatus.NOT_FOUND)
+        assert response.json == {
+            "message": "The dataset tarball named 'random_md5_string4' is not found"
+        }
```

Comment:
Thanks for the docstrings -- they help a lot (and that's good enough).

However...

> because the `ARCHIVE` tree hasn't been filled in or mocked

If we're counting on the unit test looking at the real file system and finding it empty, that's a little weak.  It would (have) be(en) better if there were an explicit mock which drove the behavior that we want to see.